src/test/java/com/example/gymApp/service/ProfileServiceTest.java

```diff
@@ -0,0 +1,37 @@
+package com.example.gymApp.service;
+
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ProfileServiceTest {
+
+  @Test
+  public void testGenerateUsername() {
```

Comment:
Please add some negative scenarios, for example if firstName=null or emptyString